File: TaskItApi/Entities/Task.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;

namespace TaskItApi.Entities
{
    /// <summary>
    /// Task of a group that an user can be holder of.
    /// </summary>
    public class Task
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        [Required]
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime From { get; set; }
        [Required]
        public DateTime Until { get; set; }
        [Required]
        public TaskStatus Status { get; set; }
        [Required]
        public int GroupID { get; set; }
        public Group Group { get; set; }

        public ICollection<TaskHolder> TaskHolders { get; set; }
    }

File: task-it-frontend/src/app/core/models/task.ts
export class Task {
  ID: number;
  Title: string;
  Description: string;
  Status: string;
  Until: Date;
  From: Date;
  GroupID: number;
  GroupColor: string;
  TaskHolders: TaskHolder[];

  constructor(
    ID: number,
    title: string,
    description: string,
    status: string,
    until: Date,
    from: Date,
    groupID: number,
    groupColor: string,
    taskHolders: TaskHolder[]
  ) {
    this.ID = ID;
    this.Title = title;
    this.Description = description;
    this.Status = status;
    this.Until = until;
    this.From = from;
    this.GroupID = groupID;
    this.GroupColor = groupColor;
    this.TaskHolders = taskHolders;
  }
}

export class TaskOutgoing {
    Title: string;
    Description: string;
    Until: Date;
    From: Date;
    GroupID: number;
}

export class TaskHolder {
  ID: number;
  UserID: number;
  Username: string;
  Email: string;
}
